### PR TITLE
Add Kyma Binding pipelines docs

### PR DESCRIPTION
**Pull request by kjaksik**
**Description**

Changes proposed in this pull request:

- Add Kyma Binding pipelines docs

**Related issue(s)**

#9402

### CODE REVIEW

File: components/binding/docs/03-binding-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/ok-to-test sha=44fb1091e18f025a365495c2d268bce944b4239f
+```
+
+Such comment will trigger the main pipeline
+
+## [Pre-master binding k3s](https://github.com/kyma-project/kyma/blob/master/.github/workflows/pre-master-binding-k3s.yml) and [Post-master binding k3s](https://github.com/kyma-project/kyma/blob/master/.github/workflows/post-master-binding-k3s.yml)
+
+Those are the main integration jobs for the Kyma Binding component. Their responsibility is to
+
+- go test, vet, and fmt the code
```
Comment (klaudiagrz) [inline components/binding/docs/03-binding-pipelines.md:23]: Where's the verb here? 🙂

File: components/binding/docs/03-binding-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Kyma Binding Pipelines
+
+During the development of the Kyma Binding component we decided to change the pipelines from the regular jobs run on Prow. Main focus was to make them faster. The idea was to run the job on k3s, since it's objectively the [best](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use), and install only the Binding component, without the whole Kyma. The decision to use Github Actions was made since the k3s scripts were already [developed](https://www.facebook.com/groups/1345683025593684/permalink/1636584193170231/). Main problem for this approach was that GitHub doesn't propagate the secrets created on the main repository, to pull requests opened from forked repositories. As the workflow in Kyma is based on using forks we had to create a workaround for that. The idea for that was found [here](https://github.com/imjohnbo/ok-to-test). The jobs that we developed:
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply components/binding/docs/03-binding-pipelines.md:3]: I'd twist the overview part a bit and focus on the facts - simply state which infrastructure is used for this component. I'd also avoid using 2nd person plural ("we"). Let's discuss this.

File: components/binding/docs/03-binding-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## [Is trusted dev](https://github.com/kyma-project/kyma/blob/master/.github/workflows/trusted-dev.yaml)
```
[the remark this answers is not in the document]
Comment (kjaksik) [reply components/binding/docs/03-binding-pipelines.md:5]: I wanted to make it easier to see the workflow file, do you see any other way for that or should I leave it?

File: components/binding/docs/03-binding-pipelines.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kjaksik) [reply components/binding/docs/03-binding-pipelines.md:3]: Of course not, sorry it was kind of late 😅

File: components/binding/docs/03-binding-pipelines.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kjaksik) [reply components/binding/docs/03-binding-pipelines.md]: Ok, I didn't see the rest of the review, that looks much better

File: components/binding/docs/03-binding-pipelines.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Kyma Binding pipelines
+
+Kyma Binding pipelines are based on Github Actions. This decision is due to the fact that scripts to run k3s are already [developed](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh) and k3s is objectively the [best](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run on CI. Main problem for this approach was that GitHub doesn't propagate the secrets created on the main repository, to pull requests opened from forked repositories. As the workflow in Kyma is based on using forks we had to create a workaround for that. The idea for that was found [here](https://github.com/imjohnbo/ok-to-test). For this purpose, we developed the following jobs:
```
Comment (klaudiagrz) [inline components/binding/docs/03-binding-pipelines.md:3]:
```suggestion
Kyma Binding pipelines are based on Github Actions and k3s that provides [the best infrastructure](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run the CI on. You can see the script to run k3s [here](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh). As GitHub doesn't propagate the secrets created on the main repository to pull requests opened from forked repositories and the workflow in Kyma is based on using forks, we had to create a [workaround](https://github.com/imjohnbo/ok-to-test) for that. For this purpose, the following jobs were developed:
```

**Review by klaudiagrz — COMMENTED** [review]
See my suggestion and contact me if I went too far from the original context 😉

File: components/binding/docs/03-binding-pipelines.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# Kyma Binding pipelines
+
+Kyma Binding pipelines are based on Github Actions and use k3s that provides [the best infrastructure](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run the CI on. The script to run k3s is based on [this script](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh). As GitHub doesn't propagate the secrets created on the main repository to pull requests opened from forked repositories and the workflow in Kyma is based on using forks, we had to create a workaround based on [this solution](https://github.com/imjohnbo/ok-to-test). For this purpose, the following jobs were developed:
```
Comment (klaudiagrz) [inline components/binding/docs/03-binding-pipelines.md:3]:
```suggestion
Kyma Binding pipelines are based on GitHub Actions and use k3s that provides [the best infrastructure](https://github.com/kyma-incubator/local-kyma#i-see-k3s-k3d-kind-and-minikube---what-should-i-use) to run the CI on. The script to run k3s is based on [this script](https://github.com/kyma-incubator/local-kyma/blob/main/create-cluster-k3s.sh). As GitHub doesn't propagate the secrets created on the main repository to pull requests opened from forked repositories and the workflow in Kyma is based on using forks, we had to create a workaround based on [this solution](https://github.com/imjohnbo/ok-to-test). For this purpose, the following jobs were developed:
```

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**